requirePromise.js

```diff
@@ -1,7 +1,7 @@
 'use strict';
 
 module.exports = function requirePromise() {
-	if (typeof Promise !== 'function') {
+	if (typeof (Promise || global.Promise || window.Promise) !== 'function') {
```

Comment:
Yes, it will break in some situations and it demands a better fix. My fault.

However, these are not _identical ways of stating the same thing_, as shown in the image below.


<img width="820" alt="screen shot 2016-08-29 at 15 37 28" src="https://cloud.githubusercontent.com/assets/181076/18053229/b8e40a9a-6dfe-11e6-96f0-278b19319060.png">